src/main/kotlin/org/jetbrains/NestedForLoopBenchmark.kt

```diff
@@ -0,0 +1,98 @@
+package org.jetbrains
+
+import org.openjdk.jmh.annotations.*
+import org.openjdk.jmh.infra.Blackhole
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@CompilerControl(CompilerControl.Mode.DONT_INLINE)
+open class NestedForLoopBenchmark: SmallSizedBenchmark() {
+    var sizeLong: Long = 0L
+
+    @Setup
+    fun setup() {
+        sizeLong = size.toLong()
+    }
+
+    @Benchmark fun nestedIntRangeToLoop(bh: Blackhole) {
+        for (i in 1..size) {
```

Comment:
Why not test nested loops with SizedBenchmark?

Reply:
SizedBenchmark goes uses sizes 10, 1000, 100000 and using 10000 for the size meant the benchmark was running for a very long time (order of minutes per measurement) -- beyond what I think would be reasonably called a "micro" benchmark and any performance changes should be noticeable with just size 1000. SmallSizedBenchmark only uses 10 and 1000.